test/cctest/test_url.cc

```diff
@@ -0,0 +1,69 @@
+#include "node_url.h"
+#include "node_i18n.h"
+
+#include "gtest/gtest.h"
+
+using node::url::URL;
+
+class URLTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+#if defined(NODE_HAVE_I18N_SUPPORT)
+    std::string icu_data_dir;
+    node::i18n::InitializeICUDirectory(icu_data_dir);
+#endif
+  }
+
+  void TearDown() override {}
+};
+
+TEST_F(URLTest, Simple) {
+  URL simple("https://example.org:81/a/b/c?query#fragment");
+
+  CHECK_EQ(simple.protocol(), "https:");
```

Comment:
Just wondering for future reference if it is preferred to use `CHECK_EQ` instead of `EXPECT_EQ`?
The difference seems to be that CHECK_EQ will abort straight away without allowing for the rest of the tests to be run, whereas EXPECT_EQ which is non-fatal and would allow for the rest of the tests to be run and report the errors afterwards.

For example, using CHECK_EQ and forcing a check to fail:
```
[ RUN      ] URLTest.Simple
Assertion failed: ((simple.protocol()) == ("httpsss")), function TestBody, file ../test/cctest/test_url.cc, line 23.
make: *** [cctest] Abort trap: 6
```


Compared to EXPECT_EQ, and forcing a the expectation to fail:
```
[ RUN      ] URLTest.Simple
../test/cctest/test_url.cc:24: Failure
Value of: "httpss:"
Expected: simple.protocol()
Which is: "https:"
[  FAILED  ] URLTest.Simple (13 ms)
...
[  PASSED  ] 36 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] URLTest.Simple
```